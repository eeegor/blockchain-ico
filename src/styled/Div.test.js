import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Div } from '.';

describe('Div', () => {
	test('renders default div', () => {
		const tree = renderer.create(<Div>Title</Div>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('renders div with custom color', () => {
		const tree = renderer.create(<Div color="green">Title</Div>).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('color', 'green !important');
	});

	test('renders div with custom background', () => {
		const tree = renderer
			.create(<Div background="green">Title</Div>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('background', 'green !important');
	});

	test('renders div with given min height', () => {
		const tree = renderer.create(<Div minHeight="100">Title</Div>).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('min-height', '100px !important');
	});

	test('renders div with given z index', () => {
		const tree = renderer.create(<Div zIndex="100">Title</Div>).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('z-index', '100 !important');
	});
});
