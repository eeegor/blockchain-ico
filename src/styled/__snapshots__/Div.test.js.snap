// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div renders default div 1`] = `
.c0 {
  width: 100%;
}

<div
  className="c0"
>
  Title
</div>
`;

exports[`Div renders div with custom background 1`] = `
.c0 {
  width: 100%;
  background: green !important;
}

<div
  className="c0"
>
  Title
</div>
`;

exports[`Div renders div with custom color 1`] = `
.c0 {
  width: 100%;
  color: green !important;
}

<div
  className="c0"
  color="green"
>
  Title
</div>
`;

exports[`Div renders div with given min height 1`] = `
.c0 {
  width: 100%;
  min-height: 100px !important;
}

<div
  className="c0"
>
  Title
</div>
`;

exports[`Div renders div with given z index 1`] = `
.c0 {
  width: 100%;
  z-index: 100 !important;
}

<div
  className="c0"
>
  Title
</div>
`;
